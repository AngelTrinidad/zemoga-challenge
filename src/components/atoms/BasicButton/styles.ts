import {StyleSheet} from 'react-native';

import {scaleSpace} from '~helpers';

const styles = StyleSheet.create({
  button: {
    width: '100%',
    minHeight: scaleSpace(50),
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 12,
  },
});

export default styles;
