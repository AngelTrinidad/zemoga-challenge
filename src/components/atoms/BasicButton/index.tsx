import React, {FC, PropsWithChildren} from 'react';
import {TouchableOpacity, ViewStyle} from 'react-native';

import styles from './styles';
import Loader from '../Loader';

import {useTheme} from '~contexts';

export interface ButtonProps extends PropsWithChildren {
  containerStyle?: ViewStyle;
  onPress?: () => void;
  isLoading?: boolean;
}

const BasicButton: FC<ButtonProps> = ({
  children,
  containerStyle,
  onPress,
  isLoading,
}) => {
  const {colors} = useTheme();

  const themeStyles: ViewStyle = {
    backgroundColor: colors.primary,
  };

  return (
    <TouchableOpacity
      onPress={onPress}
      style={[styles.button, themeStyles, containerStyle]}>
      {isLoading ? <Loader color="white" size="small" /> : children}
    </TouchableOpacity>
  );
};

export default BasicButton;
