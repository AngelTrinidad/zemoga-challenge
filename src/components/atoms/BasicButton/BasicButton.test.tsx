import React from 'react';
import {Text} from 'react-native';

import {fireEvent} from '@testing-library/react-native';

import BasicButton from '.';

import {render} from '~helpers/testing';

describe('<BasicButton />', () => {
  let component: any;
  const onPressMock = jest.fn();

  beforeEach(() => {
    component = render(
      <BasicButton onPress={onPressMock}>
        <Text>Press</Text>
      </BasicButton>,
    );
  });

  it('should render correctly', () => {
    expect(component).toBeDefined();
  });

  it('should has same snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should call onPress callback on tapped it', () => {
    const button = component.getByText('Press');
    fireEvent(button, 'press');
    expect(onPressMock).toBeCalled();
  });
});
