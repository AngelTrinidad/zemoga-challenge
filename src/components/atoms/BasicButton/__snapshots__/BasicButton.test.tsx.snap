// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasicButton /> should has same snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#567189",
      "borderRadius": 12,
      "justifyContent": "center",
      "minHeight": 57.14285714285714,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text>
    Press
  </Text>
</View>
`;
